File: experiments/runs/2026-05-25_18-43-17_claim-office-example-mapping_v8a-delayed-refactor-agent_opus-4-7-portkey-no-thinking/src/policy.ts
import { Item, Policy } from './types.js';
import { insuranceValueOf, isKnownType } from './catalogue.js';

const CAP_MULTIPLIER = 2;

function validateItems(items: Item[]): void {
  for (const item of items) {
    if (!isKnownType(item.type)) {
      throw new Error(`Unknown item type: ${item.type}`);
    }
  }
}

export function makePolicy(items: Item[]): Policy {
  validateItems(items);
  const insuranceSum = items.reduce(
    (sum, item) => sum + insuranceValueOf(item.type),
    0
  );
  const cap = insuranceSum * CAP_MULTIPLIER;
  return { items, insuranceSum, cap, remainingCap: cap };
}

File: experiments/runs/2026-09-22_00-02-08_claim-office-python-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/policy.py
"""A policy created by a quote step: what is insured, and how much cap is left."""

from catalogue import insurance_value

CAP_MULTIPLIER = 2


def insurance_sum(items):
    return sum(insurance_value(item["type"]) for item in items)


def payout_cap(items):
    return CAP_MULTIPLIER * insurance_sum(items)
